llama-index-core/llama_index/core/llms/callbacks.py

```diff
@@ -236,25 +236,34 @@ def wrapper_logic(_self: Any) -> Generator[CallbackManager, None, None]:
 
             yield callback_manager
 
+        def extract_prompt(*args: Any, **kwargs: Any) -> str:
+            if len(args) > 0:
+                return str(args[0])
+            elif "prompt" in kwargs:
+                return kwargs["prompt"]
+            else:
+                return str(kwargs)
```

Comment:
Why we want to cast `kwargs` to str? 🤔

Reply:
Logan suggested this, I guess it is just one way to avoid an exception. The obvious alternative is raising a ValueError or some other exception. I think that might be more appropriate, since this else block covers unexpected / improper usage, but let me know what you think!

Reply:
I would vote for ValueError 🙋  
```suggestion
raise ValueError("No prompt provided in positional or keyword arguments")
```